packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -91,7 +91,19 @@ export class DefaultCrudRepository<T extends Entity, ID>
       `Entity ${entityClass.name} must have at least one id/pk property.`,
     );
 
-    // Create an internal legacy Model attached to the datasource
+    this.setupPersistedModel(definition);
+  }
+
+  // Create an internal legacy Model attached to the datasource
+  private setupPersistedModel(definition: ModelDefinition) {
+    const dataSource = this.dataSource;
+
+    const model = dataSource.modelBuilder.models[definition.name];
+    if (model) {
+      // The backing persisted model has been already defined.
+      this.modelClass = model as typeof juggler.PersistedModel;
+      return;
+    }
```

Comment:
There is a flaw in this PR:

1. A 3.x model classes can only be attached to one data source.
2. We introduce `Repository` so that it can represent `modelClass + dataSource`
3. Each instance of `Repository` should have its own 3.x model class 
4. We should use `bindModel` instead of `model.attachTo()`

Reply:
The 3.x model classes maintain two states:
1. The model definition
2. The attached data source

For example, a `Customer` model can only be attached to `mongodb` or `mysql` but we cannot have the same model definition to be used with `mongodb` or `mysql` unless we subclass it.

In LB4, we subclass `Customer` for each repository instance to delegate to juggler model methods without interfering with other instances. For example:

```ts
export class MyController {
  @repository(Customer, 'mongodb')
  private mongoRepo;

  @repository(Customer, 'mysql')
  private mysqlRepo;
}
```

@bajtos Please note in LB 3.x, all data source instances share the same `modelBuilder` if it's from `loopback` module. See https://github.com/strongloop/loopback/blob/master/lib/registry.js#L362.

If in LB4, we always create DataSource instances with its own modelBuilder, your PR should be fine.

Reply:
> If in LB4, we always create DataSource instances with its own modelBuilder, your PR should be fine.

Yes, that's the case so far - in LB4, each DataSource has its own model builder.